code/game/machinery/recycle_vendor.dm

```diff
@@ -0,0 +1,285 @@
+/obj/machinery/recycle_vendor
+	name = "recycling vendor"
+	desc = "Recycle today for a better tomorrow!"
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "recycle"
+	layer = BELOW_OBJ_LAYER
+	anchored = TRUE
+	density = TRUE
+	use_power = IDLE_POWER_USE
+
+	// Can't use subtypeof(), since we have lots of retarded materials
+	var/list/materials_supported = list(
+		MATERIAL_STEEL,
+		MATERIAL_GLASS,
+		MATERIAL_PLASTIC,
+		MATERIAL_WOOD,
+		MATERIAL_SILVER,
+		MATERIAL_GOLD,
+		MATERIAL_URANIUM,
+		MATERIAL_CARDBOARD,
+		MATERIAL_PLASMA,
+		MATERIAL_PLATINUM,
+		MATERIAL_PLASTEEL,
+		MATERIAL_DIAMOND)
+	var/list/materials_allowed = list()
+	var/list/materials_stored = list()
+	var/list/stored_item_materials = list()
+	var/obj/item/stored_item_object = null
+	var/stored_item_value = 0
+	var/stored_item_fluff = ""
+	var/sales_paused = FALSE
+	var/vagabond_charity_budget = 2000
+	var/datum/money_account/merchants_pocket
+	var/datum/wires/recycle_vendor/wires
+	var/wire_flags = 0
+
+
+/obj/machinery/recycle_vendor/New(loc, ...)
+	. = ..()
+	materials_allowed = materials_supported.Copy()
+	wires = new(src)
+	update_icon()
+
+
+/obj/machinery/recycle_vendor/Destroy()
+	qdel(wires)
+	eject_stored_materials()
+	eject_stored_item()
+	..()
+
+
+/obj/machinery/recycle_vendor/update_icon()
+	SSnano.update_uis(src)
+	overlays.Cut()
+	if(stat & BROKEN)
+		icon_state = "recycle_broken"
+		return
+
+	icon_state = "recycle"
+
+	if(stat & NOPOWER || !anchored)
+		return
+
+	overlays += sales_paused || !materials_allowed.len	? "recycle_screen_red"			: "recucle_screen_green"
```

Comment:
I was going to say you got a typo but you actually also named the overlay "recucle" so I guess it checks out.

Reply:
Yeah, I fat fingered it.